src/protocol/mod.rs

```diff
@@ -41,8 +42,15 @@ impl Handshake {
 pub enum Request {
     /// A get request for a blob or collection
     Get(GetRequest),
+    /// A get request that allows the receiver to create a collection
+    CustomGet(Bytes),
```

Comment:
Is it bikeshedding time?  I vote `Query`.  But never mind, I vote not slowing down because of bikeshedding.

Reply:
Query is a bit generic, since we are going to have a Query request for the discovery part. E.g. what children of this hash do you actually have...

My rationale for the naming is that this is still fundamentally a get request, just a named one, not a content-addressed one.